1324 - Calendar improvements and add/remove/delete API

tmcconechy opened the pull request:
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

**Related github/jira issue (required)**:
Closes #1324 
Closes #1436

**Steps necessary to review your pull request (required)**:

**Adding Variations**
http://localhost:4000/components/calendar/example-index.html
- in the plus menu on the top use the three options
- click a date 
- first menu option adds one directly with the api
- second menu option adds one directly with the api in a modal
- third menu option tests the clear all events api


**Showing Multiple**
http://localhost:4000/components/calendar/example-index.html
- in the plus menu on the top use the three options
- click a date like Jan 20 with several events
- look at the accordion on the right
- can also do the add a few times in the menu to add a few to the same day and show that

**Click operations**
http://localhost:4000/components/calendar/example-index.html
- click an date with an event and it should show on the right
- click an event and you should be able to edit it in a modal
- dbl click on a date and you will be able to add an event to that date.

**1436**
- follow the steps again on https://github.com/infor-design/enterprise/issues/1436
- this is now fixed

**popover changes**
- cleaned up listing while testing http://localhost:4000/components/popover/list
- added an option to show a header color http://localhost:4000/components/popover/example-header-color.html

**icons**
http://localhost:4000/components/alerts/example-index.html
- subtle but changed the svg to have a white icon in the background, before it was transparent and wouldnt show up well.
- you can see the icon on http://localhost:4000/components/calendar/example-index.html (Feb 10th click on the day

tmcconechy commented:
- Is it weird there's no submit button just an X (this is true when adding any event via modal (e.g., double clicking on a day):

Maybe, i thought it was ok as you can also click out to save it. It just auto saves whenever the dialog closes. Thought this was maybe easier than pressing ok/cancel. Willing to see what others say, other part of it is this is a popup not a modal so no buttons there at the moment so this was a lot easier. 

- This shows for me on the right (I don't see anything in Upcoming Events at any point)
popover changes

Sorry i meant shows on the right side. Will correct that but if you clicked on for example a day with 14days in the calendar of today and add an event then it will show on the left in upcoming events.

- Do you mean the header of the popover? If so, I'm not seeing an option to change this
I mean this example http://localhost:4000/components/popover/example-extra-css-class.html (if one was to change that class in the example the color would change). Also you click on different events to edit you can see the header change on the popup to match.

deep7102 commented:
Looks good! the thing I found as
Adding Variations
1) Getting error
- in the plus menu on the top use the option "clear all"
- then use the option "Add Event (Api)"
- See the console, it show the error

2) Second menu option "Add Event (Modal)"
- My vote go with @davidcarlsonberg adding submit button, which can cancel if X clicked

tmcconechy commented:
OK, Fixed a couple comments.

1. Fixed an error after clear events as noted by @deep7102 
2. Made modal have buttons for cancel/submit as noted by everyone
3. Fixed a bug with empty subjects - now if you dont type a subject on the modal it defaults to (No Title)
